airflow/contrib/hooks/nomad_hook.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+""" Hook to manage connection to nomad server
+    NOTE: this operator also relies on the python-nomad
+          package https://github.com/jrxFive/python-nomad """
+
+import nomad
+
+from airflow.hooks.base_hook import BaseHook
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class NomadHook(BaseHook, LoggingMixin):
+    def __init__(self,
+                 nomad_conn_id='nomad_default',
+                 *args,
+                 **kwargs):
+        self.connection = self.get_connection(nomad_conn_id)
+        self.nomad_client = self.get_nomad_client(*args, **kwargs)
+
+    def get_nomad_client(self, *args, **kwargs):
```

Comment:
This one should override `get_conn`, please refer to other hooks, for example: https://github.com/apache/incubator-airflow/blob/master/airflow/hooks/http_hook.py

Reply:
@Fokko what do you mean  get_nomad_client  returns a client and not a connection do I need get_conn anyway ?  thanks

Reply:
But the client represents a connection, right?

Reply:
@Fokko ?

Reply:
From what I understand from a quick check, the `get_conn` seems to be intended to be the convention for retrieving a connection for the an external resource, and it typically uses an airflow connection to supply the connection details.  It appears that the proposed function here would does that, it's just named `get_nomad_client` instead.

In this particular case, it seems that simply renaming `get_nomad_client` to `get_conn` would be sufficient. Does that sound right?